### PR TITLE
Use babel preset env

### PR DESCRIPTION
**Pull request by calthomson**
## Status
**DOING FINAL TESTING**

## Description
* Replaced `babel-preset-esXXXX` npm packages with `babel-preset-env`. [babel-preset-env info](https://babeljs.io/env/)
* Added `babel-loader` because it seems like `babel-preset-env` requires it.
* Got rid of plugins that aren't actually used by our projects, and the tests for those plugins. (we don't seem to use the Javascript `do` expression anywhere.

## Impacted Areas in Application
* `index.js`
* `package.json`

### CODE REVIEW

File: tests/input.js [hunk 1]
```diff
@@ -11,26 +11,11 @@ let { x, y, ...z } = { x: 1, y: 2, a: 3, b: 4 };
...
 function dec(id){
   return (target, property, descriptor) => console.log('executed', id);
 }
 
-// class decorators and properties
-class Foobar {
-  @dec(1)
-  static foo = 1;
-}
-
```
Comment (samprity) [inline tests/input.js]: What was all of these ? Random tests?

File: package.json [hunk 2]
```diff
@@ -20,23 +20,20 @@
   "main": "index.js",
   "dependencies": {
     "babel-plugin-transform-class-properties": "^6.19.0",
-    "babel-plugin-transform-decorators-legacy": "^1.3.4",
-    "babel-plugin-transform-do-expressions": "^6.8.0",
     "babel-plugin-transform-object-rest-spread": "^6.19.0",
-    "babel-preset-es2015": "^6.24.0",
```
Comment (ForceKinetics) [inline package.json]: Wow these just kept getting appended? So crazy

File: tests/input.js (hunk 1 above)
Comment (calthomson) [reply tests/input.js]: So we had these plugins to allow us to do certain things in JS that, as far as I can tell, we don't actually do in our projects. These tests are from the first commit of this repo and I don't know if they were actually added thoughtfully or if they were just copy&pasted from somewhere else, so I thought I'd cut the fat ✂️

Comment (calthomson) [reply tests/input.js]: Before I commit I'll make doubly sure we don't use those plugins - I know they're not in Appearance-Extension or Phoenix-Client

**Comment by calthomson** [comment]
If I change one of our projects, say Appearance-Extension to use babel-preset-env and all the required transformation plugins, and remove babel-preset-ca, it all works, but as soon as I outsource the babel packages to babel-preset-ca, and import that to a project, I get this error:

screenshot 2018-02-01 09 36 11

Comparing the results of `npm-ls` for babel-preset-env and Appearance-Extension (with the babel-preset-env changes) it looks like all the dependencies and their versions are the same. Wondering if the `.babelrc` config is somehow getting messed up.

**Pull request merged**